File: Find the minimum time - GFG/find-the-minimum-time.cpp
// { Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
// User function Template for C++

class Solution{
public:
    int minTime(int S1, int S2, int N){
    int maxcount=INT_MAX;
       if(S1*N<=S2*N){
           maxcount=min(S1*N,maxcount);
       }
       else{
           maxcount=min(S2*N,maxcount);
       }
       for(int i=0;i<=N;i++){
           int count=max(S1*i,S2*(N-i));
           if(maxcount>count){
               maxcount=count;
           }
       }
       return maxcount;
    }
};

// { Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int S1, S2, N;
        cin>>S1>>S2>>N;
        
        Solution ob;
        cout<<ob.minTime(S1, S2, N)<<"\n";
    }
    return 0;
}  // } Driver Code Ends

File: Find the minimum time - GFG/find-the-minimum-time.java
// { Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
import java.lang.*;

class GFG{
    public static void main(String args[])throws IOException
    {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(in.readLine());
        while(t-- > 0){
            String arr[] = in.readLine().trim().split("\\s+");
            int S1 = Integer.parseInt(arr[0]);
            int S2 = Integer.parseInt(arr[1]);
            int N = Integer.parseInt(arr[2]);
            
            Solution ob = new Solution();
            System.out.println(ob.minTime(S1, S2, N));
        }
    }
}// } Driver Code Ends


//User function Template for Java

class Solution{
    static int minTime(int S1, int S2, int N)
    {
        int maxCount = Integer.MAX_VALUE;
        int count=0;
        if(S1*N<=S2*N)
        {
            maxCount = Integer.min(S1*N,maxCount);
        }
        else
        {
            maxCount = Integer.min(S2*N,maxCount);
        }
        for(int i=0;i<=N;i++)
        {
            count = Integer.max(S1*i,S2*(N-i));
            if(maxCount>count)
            {
                maxCount=count;
            }
        }
        return maxCount;
    }
}